Expose additional winrm options

frezbo opened the pull request:
Signed-off-by: Noel Georgi <git@frezbo.com>

Fixes: #391

frezbo commented:
rerunning to fix the GCP fluke.

frezbo commented:
The failure is related to GCP test suite.

clintoncwolfe commented on lib/train/transports/winrm.rb:
```diff
@@ -45,6 +45,9 @@ class WinRM < Train.plugin(1)
     option :port
     option :user, default: 'administrator', required: true
     option :password, nil
+    option :transport, default: :negotiate
```
For now, let's add a --winrm- prefix here, to match what InSpec will be sending.  I think in the future we may be able to something like having InSpec strip the prefix from options (or all options would have the prefix - point being it should be consistent)

frezbo replied:
resolved.

clintoncwolfe commented on lib/train/transports/winrm.rb:
```diff
@@ -100,9 +103,9 @@ def validate_options(opts)
-        transport:                :negotiate,
-        disable_sspi:             false,
-        basic_auth_only:          false,
+        transport:                opts[:transport].intern,
```
It's equivalent, but more readable, to say `.to_sym` here.

clintoncwolfe replied:
Actually, come to think of it, we might want to add some validation here; I'm not sure what valid values are for the transport flag.

frezbo replied:
resolved.

clintoncwolfe commented on lib/train/transports/winrm.rb:
```diff
@@ -45,6 +45,9 @@ class WinRM < Train.plugin(1)
     option :port
     option :user, default: 'administrator', required: true
     option :password, nil
+    option :transport, default: :negotiate
```
It would be great if you could add a test file in `test/windows/` to exercise the new options (those tests are run by `rake test:windows`, see Rakefile for details).  If that would be overly difficult to do, I'm OK with omitting that for now - but please open a new issue on Train mentioning the need to add testing for the new winrm options, so it doesn't get lost.

frezbo replied:
resolved.

frezbo commented:
@clintoncwolfe I have updated the PR as per the review. The tests are failing due to `gcp` UT's failing. (It is also faky on the local machine, would be good to fix that). I have omitted `kerberos` transport. The `ssl` transport also exposes some additional options, but it would go out of scope of this MVP. I think it would be good to track that later. I have also added a baseline UT's for the winrm transport.

frezbo commented:
Re-trigger tests

clintoncwolfe commented:
Disregarding unrelated failing unit test per #393
